File: minimum_jerk_trajectory_planner/minimum_jerk_trajectory_planner/Pose.py
class Pose:
    def __init__(self, x=0, y=0, theta=0):
        self.x = x
        self.y = y
        self.theta = theta
    
    def __sub__(self, other):
        return Pose(self.x - other.x,
                    self.y - other.y,
                    self.theta - other.theta)

File: minimum_jerk_trajectory_planner/src/pose.cpp
#include "minimum_jerk_trajectory_planner/pose.hpp"

namespace minimum_jerk
{
    Pose::Pose(double px, double py, double ptheta)
    {
        x_ = px;
        y_ = py;
        theta_ = ptheta;
    }
    Pose Pose::operator-(Pose other)
    {
        return Pose(this->x_ - other.x_, this->y_ - other.y_, this->theta_ - other.theta_);
    }
    bool Pose::operator==(Pose other)
    {
        return (this->x_ == other.x_ && this->y_ == other.y_ && this->theta_ == other.theta_);
    }
    double Pose::get_x() const
    {
        return x_;
    }
    double Pose::get_y() const
    {
        return y_;
    }
    double Pose::get_theta() const
    {
        return theta_;
    }
}